packages/desktop-client/src/components/reports/spreadsheets/net-worth-spreadsheet.ts

```
@@ -61,27 +67,61 @@
+        if (interval === 'Weekly') {
+          // Group transactions by week and sum their amounts
+          const weeklyBalances: Record<string, number> = {};
+          balances.forEach(b => {
+            const weekDate = monthUtils.weekFromDate(b.date, firstDayOfWeekIdx);
+            weeklyBalances[weekDate] = (weeklyBalances[weekDate] || 0) + b.amount;
+          });
+
+          // Convert back to Balance format
+          processedBalances = Object.entries(weeklyBalances).reduce(
+            (acc, [date, amount]) => ({
+              ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)